### PR TITLE
fix location

### PR DESCRIPTION
**Pull request by Amoki**
## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [ ] My code follows the code style of this project.
- [x] I have tested my code.
- [x] I want to run the tests for the commits of this PR

### CODE REVIEW

**Review by Kurtil — APPROVED** [review]

File: src/utils/location.js [hunk 1]
```diff
@@ -2,37 +2,82 @@
...
+ * @param {Number} param longitude DD coordinate
+ * @returns {[Array, Array]} latitude and longitude DMS coordinate
  */
-function DD2DMS(latOrLong) {
-  const result = new Array(3);
-  result[0] = 0 | latOrLong;
-  if (latOrLong < 0) {
-    latOrLong *= -1;
+function DD2DMS(lat, long) {
+  console.log("DD2DMS:", lat, long)
```
Comment (NicolasRichel) [inline src/utils/location.js]: Looks like your left some dev logs. ;)

File: src/utils/location.js [hunk 2]
```diff
@@ -2,37 +2,82 @@
...
+  const dmsCoords = new DmsCoordinates(lat, long);
+  const { longitude, latitude } = dmsCoords.dmsArrays;
+  let [latD, latM, latS, latDir] = latitude;
+  console.log("latitude:", [latD, latM, latS, latDir])
```
Comment (NicolasRichel) [inline src/utils/location.js]: dev logs ?

File: src/utils/location.js [hunk 3]
```diff
@@ -2,37 +2,82 @@
...
+  if (latDir == "S") {
+    latD *= -1;
+    latM *= -1;
+    latS *= -1;
+  }
+
+  let [longD, longM, longS, longDir] = longitude;
+  console.log("longitude:", [longD, longM, longS, longDir])
```
Comment (NicolasRichel) [inline src/utils/location.js]: dev logs ?

File: src/utils/location.js [hunk 4]
```diff
@@ -2,37 +2,82 @@
  * Coordinate systems:
  *   - DMS: Degrees Minutes Seconds, example: 48° 51' 30.24" N, 2° 17' 40.2" E
  *   - DD: Decimal Degrees, example: 48.8584° N, 2.2945° E
+ *
+ * IFC uses a sightly different format for DD and DMS:
+ *   DMS is an array of 3 in IFC2x3 or 3 to 4 in IFC 4+. The first element is the degrees, the second is the minutes the third is the seconds, the fourth is the number of millionth-seconds.
+ *   When the direction is West or South, all components me negative. When direction is East or North, all components are negatives.
```
Comment (NicolasRichel) [inline src/utils/location.js]: Shouldn't that sentence be:
> When the direction is West or South, all components **are** negative. When direction is East or North, all components are **positive**.

**Review by NicolasRichel — CHANGES_REQUESTED** [review]

**Review by NicolasRichel — APPROVED** [review]

**Pull request merged**